From 430adaf3dbc270f8825c137cfb6e93e49b208078 Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Thu, 12 Jun 2025 07:47:36 +0200
Subject: [PATCH] pyproject.toml: add file

as setuptools 80.x deprecated easy_install module,
so setup.py can't be used after the deadline of end
of October 2025.
Add a pyproject.toml created using setuptools-py2cfg
and ini2toml with manually added build-system
section

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
Upstream-Status: Submitted [https://github.com/a13xp0p0v/kernel-hardening-checker/pull/193]
---
 pyproject.toml | 39 +++++++++++++++++++++++++++++++++++++++
 setup.cfg      | 32 --------------------------------
 setup.py       | 15 ---------------
 3 files changed, 39 insertions(+), 47 deletions(-)
 create mode 100644 pyproject.toml
 delete mode 100644 setup.cfg
 delete mode 100755 setup.py

diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000..74a625e
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,39 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[metadata]
+name = "kernel-hardening-checker"
+version = "attr: kernel_hardening_checker.__version__"
+author = "Alexander Popov"
+author_email = "alex.popov@linux.com"
+home_page = "https://github.com/a13xp0p0v/kernel-hardening-checker"
+description = "A tool for checking the security hardening options of the Linux kernel"
+long_description = "file: README.md"
+license = "GNU General Public License v3 (GPLv3)"
+license_files = "LICENSE.txt"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+    "Topic :: Security",
+    "Operating System :: POSIX :: Linux",
+    "Environment :: Console",
+    "Programming Language :: Python :: 3",
+]
+
+[options]
+packages = [
+    "kernel_hardening_checker",
+    "kernel_hardening_checker.config_files",
+    "kernel_hardening_checker.config_files.defconfigs",
+    "kernel_hardening_checker.config_files.distros",
+    "kernel_hardening_checker.config_files.kspp-recommendations",
+]
+include_package_data = true
+
+[project]
+name = "kernel-hardening-checker"
+dynamic = ["version"]
+
+[project.scripts]
+kernel-hardening-checker = "kernel_hardening_checker:main"
diff --git a/setup.cfg b/setup.cfg
deleted file mode 100644
index bfe5105..0000000
--- a/setup.cfg
+++ /dev/null
@@ -1,32 +0,0 @@
-[metadata]
-name = kernel-hardening-checker
-version = attr: kernel_hardening_checker.__version__
-author = Alexander Popov
-author_email = alex.popov@linux.com
-home_page = https://github.com/a13xp0p0v/kernel-hardening-checker
-description = A tool for checking the security hardening options of the Linux kernel
-long_description = file: README.md
-license = GNU General Public License v3 (GPLv3)
-license_files = LICENSE.txt
-classifiers =
-	Development Status :: 5 - Production/Stable
-	License :: OSI Approved :: GNU General Public License v3 (GPLv3)
-	Topic :: Security
-	Operating System :: POSIX :: Linux
-	Environment :: Console
-	Programming Language :: Python :: 3
-
-[options]
-setup_requires = setuptools
-packages =
-	kernel_hardening_checker
-	kernel_hardening_checker.config_files
-	kernel_hardening_checker.config_files.defconfigs
-	kernel_hardening_checker.config_files.distros
-	kernel_hardening_checker.config_files.kspp-recommendations
-include_package_data = true
-
-[options.entry_points]
-console_scripts =
-	kernel-hardening-checker = kernel_hardening_checker:main
-
diff --git a/setup.py b/setup.py
deleted file mode 100755
index f8fac99..0000000
--- a/setup.py
+++ /dev/null
@@ -1,15 +0,0 @@
-#!/usr/bin/env python3
-
-"""
-This tool is for checking the security hardening options of the Linux kernel.
-
-SPDX-FileCopyrightText: Alexander Popov <alex.popov@linux.com>
-SPDX-License-Identifier: GPL-3.0-only
-
-This module performs installing of the kernel-hardening-checker package.
-"""
-
-from setuptools import setup
-
-# See the options in setup.cfg
-setup()
-- 
2.34.1
